lib/platform/linux-i2c.c

```diff
@@ -351,6 +417,9 @@ struct switchtec_dev *switchtec_open_i2c(const char *path, int i2c_addr)
 	if (i2c_set_addr(idev, i2c_addr))
 		goto err_close_free;
 
+	if (i2c_gas_cap_get(&idev->dev) != GAS_TWI_MODE_MASK)
+		goto err_close_free;
+
```

Comment:
So do we not support the legacy TWI mode? If so please not that in the commit message.

Reply:
The Switchtec firmware still supports legacy TWI mode, but it is not suggested. So I do not support the legacy TWI mode in switchtec-user. I wrote it in commit message.